tests/test_plugins/test_plugins.py

```diff
@@ -0,0 +1,273 @@
+                plugin_name=plugin_name,
+                plugin_entrypoint=entry_point,
+            )
+        )
+    command = [
+        python_version,
+        "-m",
+        "pip",
+        "install",
+        str(plugin_dir.absolute()),
+    ]
+    out, err, exit_code = capture(command, cwd=str(plugin_dir))
+    print(out)
+    assert exit_code == 0, err
+    yield module_dir
+    shutil.rmtree(plugin_dir)
+    command = [python_version, "-m", "pip", "uninstall", plugin_name, "-y"]
+    out, err, exit_code = capture(command)
+    print(out)
+    assert exit_code == 0, err
+
+
+@pytest.mark.slow
+def test_plugin_no_all(tmp_path, module_no_all_plugin, python_version):
+    cfg_file = cfg_file_create(cfg_file_contents, tmp_path)
+    scene_name = "NoAllTest"
+    command = [
+        python_version,
+        "-m",
+        "manim",
+        simple_scenes_path(),
+        scene_name,
+        "-ql",
+        "--media_dir",
+        str(cfg_file.parent),
+        "--config_file",
+        str(cfg_file),
+    ]
+    out, err, exit_code = capture(command, cwd=str(cfg_file.parent))
+    print(out)
+    print(err)
+    assert exit_code == 0, err
+
+
+@pytest.fixture
+def module_with_all_plugin(tmp_path, python_version):
+    plugin_name = "test_plugin"
+    entry_point = f"{plugin_name}"
+    plugin_dir = tmp_path / "module_entrypoint_with_all"
+    module_dir = plugin_dir / plugin_name
+    module_dir.mkdir(parents=True)
+    with open(module_dir / "__init__.py", "w") as f:
+        f.write("__all__=['WithAll']\n")
+        f.write(
+            plugin_init_template.format(
+                class_name="WithAll",
+                function_name="import_all",
+            )
+        )
+    with open(plugin_dir / "pyproject.toml", "w") as f:
+        f.write(
+            plugin_pyproject_template.format(
+                plugin_name=plugin_name,
+                plugin_entrypoint=entry_point,
+            )
+        )
+    command = [
+        python_version,
+        "-m",
+        "pip",
+        "install",
+        str(plugin_dir.absolute()),
+    ]
+    out, err, exit_code = capture(command, cwd=str(plugin_dir))
+    print(out)
+    assert exit_code == 0, err
+    yield module_dir
+    shutil.rmtree(plugin_dir)
+    command = [python_version, "-m", "pip", "uninstall", plugin_name, "-y"]
+    out, err, exit_code = capture(command)
+    print(out)
+    assert exit_code == 0, err
+
+
+@pytest.mark.slow
+def test_plugin_with_all(tmp_path, module_with_all_plugin, python_version):
+    cfg_file = cfg_file_create(cfg_file_contents, tmp_path)
+    scene_name = "WithAllTest"
+    command = [
+        python_version,
+        "-m",
+        "manim",
+        simple_scenes_path(),
+        scene_name,
+        "-ql",
+        "--media_dir",
+        str(cfg_file.parent),
+        "--config_file",
+        str(cfg_file),
+    ]
+    out, err, exit_code = capture(command, cwd=str(cfg_file.parent))
+    print(out)
+    print(err)
```

Comment:
Nitpick, question: do we always want to print this out every time we run the test suite?

Reply:
Usually, it will not be printed using `-s` is passed to `pytest`, and prints while it fails. Doesn't hurt to have it.